### ISSUE TITLE
Thumbnails not generating when using with AWS S3 + heroku

### ISSUE THREAD

**Issue by erichonkanen** [SmileyChris/easy-thumbnails#386]
Hello,

I'm using this package on py3.4 / dj 1.8 project and for some reason the thumbnails are not being generated once I switched to using amazon s3 with heroku.

# settings.py
```
# AWS - http://developer.yahoo.com/performance/rules.html#expires
AWS_HEADERS = {
    'Expires': 'Thu, 31 Dec 2099 20:00:00 GMT',
    'Cache-Control': 'max-age=94608000',
}
AWS_STORAGE_BUCKET_NAME = 'xxx'
AWS_ACCESS_KEY_ID = 'xxx''
AWS_SECRET_ACCESS_KEY = 'xxx'
AWS_S3_CUSTOM_DOMAIN = '%s.s3.amazonaws.com' % AWS_STORAGE_BUCKET_NAME
STATICFILES_LOCATION = 'static'
STATICFILES_STORAGE = 'craft.settings.storages.StaticStorage'
STATIC_URL = "https://%s/%s/" % (AWS_S3_CUSTOM_DOMAIN, STATICFILES_LOCATION)
MEDIAFILES_LOCATION = 'media'
MEDIA_URL = "https://%s/%s/" % (AWS_S3_CUSTOM_DOMAIN, MEDIAFILES_LOCATION)
DEFAULT_FILE_STORAGE = 'craft.settings.storages.MediaStorage'

# easy_thumbnails
THUMBNAIL_DEFAULT_STORAGE = DEFAULT_FILE_STORAGE
```

Here is my custom Storage class
#storages.py
```
from django.conf import settings
from storages.backends.s3boto import S3BotoStorage


class StaticStorage(S3BotoStorage):
    location = settings.STATICFILES_LOCATION


class MediaStorage(S3BotoStorage):
    location = settings.MEDIAFILES_LOCATION
```

any help appreciated..

`Foo.image.get_thumbnail_url('default')`  == ''
`Foo.image.url` == http://amazon_url (working)

**Comment by erichonkanen** [comment]
Update: After much head-scratching and headache I was able to discover an error was occuring relating to `select_related`

```
In [15]: f.file.get_thumbnail(thumbnail_options={'size': (100, 100), 'crop': True})
---------------------------------------------------------------------------
FieldError                                Traceback (most recent call last)
<ipython-input-15-3e15d621ece7> in <module>()
----> 1 f.file.get_thumbnail(thumbnail_options={'size': (100, 100), 'crop': True})

/app/.heroku/python/lib/python3.4/site-packages/easy_thumbnails/files.py in get_thumbnail(self, thumbnail_options, save, generate, silent_template_exception)
    507             generate = self.generate
    508
--> 509         thumbnail = self.get_existing_thumbnail(thumbnail_options)
    510         if not thumbnail:
    511             if generate:

/app/.heroku/python/lib/python3.4/site-packages/easy_thumbnails/files.py in get_existing_thumbnail(self, thumbnail_options, high_resolution)
    473
    474         for filename in names:
--> 475             exists = self.thumbnail_exists(filename)
    476             if exists:
    477                 thumbnail_file = ThumbnailFile(

/app/.heroku/python/lib/python3.4/site-packages/easy_thumbnails/files.py in thumbnail_exists(self, thumbnail_name)
    588                 self.source_storage, self.name)
    589         else:
--> 590             source = self.get_source_cache()
    591             if not source:
    592                 return False

/app/.heroku/python/lib/python3.4/site-packages/easy_thumbnails/files.py in get_source_cache(self, create, update)
    623             create=create, update_modified=update_modified,
    624             storage=self.source_storage, name=self.name,
--> 625             check_cache_miss=self.thumbnail_check_cache_miss)
    626         return self._source_cache
    627

/app/.heroku/python/lib/python3.4/site-packages/easy_thumbnails/models.py in get_file(self, storage, name, create, update_modified, check_cache_miss, **kwargs)
     24             try:
     25                 manager = self._get_thumbnail_manager()
---> 26                 obj = manager.get(**kwargs)
     27             except self.model.DoesNotExist:
     28

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/query.py in get(self, *args, **kwargs)
    326         if self.query.can_filter():
    327             clone = clone.order_by()
--> 328         num = len(clone)
    329         if num == 1:
    330             return clone._result_cache[0]

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/query.py in __len__(self)
    142
    143     def __len__(self):
--> 144         self._fetch_all()
    145         return len(self._result_cache)
    146

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/query.py in _fetch_all(self)
    963     def _fetch_all(self):
    964         if self._result_cache is None:
--> 965             self._result_cache = list(self.iterator())
    966         if self._prefetch_related_lookups and not self._prefetch_done:
    967             self._prefetch_related_objects()

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/query.py in iterator(self)
    236         # Execute the query. This will also fill compiler.select, klass_info,
    237         # and annotations.
--> 238         results = compiler.execute_sql()
    239         select, klass_info, annotation_col_map = (compiler.select, compiler.klass_info,
    240                                                   compiler.annotation_col_map)

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/sql/compiler.py in execute_sql(self, result_type)
    827             result_type = NO_RESULTS
    828         try:
--> 829             sql, params = self.as_sql()
    830             if not sql:
    831                 raise EmptyResultSet

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/sql/compiler.py in as_sql(self, with_limits, with_col_aliases, subquery)
    376         refcounts_before = self.query.alias_refcount.copy()
    377         try:
--> 378             extra_select, order_by, group_by = self.pre_sql_setup()
    379             if with_limits and self.query.low_mark == self.query.high_mark:
    380                 return '', ()

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/sql/compiler.py in pre_sql_setup(self)
     46         might not have all the pieces in place at that time.
     47         """
---> 48         self.setup_query()
     49         order_by = self.get_order_by()
     50         extra_select = self.get_extra_select(order_by, self.select)

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/sql/compiler.py in setup_query(self)
     37         if all(self.query.alias_refcount[a] == 0 for a in self.query.tables):
     38             self.query.get_initial_alias()
---> 39         self.select, self.klass_info, self.annotation_col_map = self.get_select()
     40         self.col_count = len(self.select)
     41

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/sql/compiler.py in get_select(self)
    204
    205         if self.query.select_related:
--> 206             related_klass_infos = self.get_related_selections(select)
    207             klass_info['related_klass_infos'] = related_klass_infos
    208

/app/.heroku/python/lib/python3.4/site-packages/django/db/models/sql/compiler.py in get_related_selections(self, select, opts, root_alias, cur_depth, requested, restricted)
    752                     'Choices are: %s' % (
    753                         ', '.join(invalid_fields),
--> 754                         ', '.join(_get_field_choices()) or '(none)',
    755                     )
    756                 )

FieldError: Invalid field name(s) given in select_related: 'dimensions'. Choices are: (none)
```

by setting `THUMBNAIL_CACHE_DIMENSIONS = False`, it resolved and works properly uploading to amazon s3

maybe this should be documented and better handled?

**Comment by SmileyChris** [comment]
Apologies for the bug -- this is fixed in #369, release coming soon
